py/server/deephaven/exec_ctx.py

```diff
@@ -0,0 +1,63 @@
+#
+#     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+"""This module gives users the ability to directly manage the Deephaven query execution context on threads, which is
+critical for applications to correctly launch deferred query evaluations, such as table update operations in threads.
+"""
+from __future__ import annotations
+
+import jpy
+
+from deephaven._wrapper import JObjectWrapper
+
+_JExecutionContext = jpy.get_type("io.deephaven.engine.context.ExecutionContext")
+
+
+class ExecutionContext(JObjectWrapper):
+    """An ExecutionContext represents a specific combination of query library, query compiler, and query scope under
+    which a query is evaluated.
+
+    A default, systemic ExecutionContext is created when a Deephaven script session is initialized, and it is used to
+    evaluate queries submitted through this script session. However, to be able to evaluate a query in a deferred
+    manner, such as in a different thread from the script session thread, an application must explicitly obtain or
+    build a ExecutionContext and either install it on the thread or use it as a context manager to enclose the query
+    in the body of the with statement.
+
+    Note that, a ExecutionContext can be shared among many threads. The most typical use pattern would be to obtain
+    the script session's systemic ExecutionContext and set it on a thread explicitly created by the user or simply
+    use it to wrap a query run in that thread.
```

Comment:
1. Python scoping rules apply to threads as well
2. Since query scope and compiler tools are not something we allow users to access, the only possible issue is with sharing the the query library among threads, but even that is a very unlikely case
3. EC is thread-local, so locking should interfere with using it